pkg/database/statistics.go

```diff
@@ -158,6 +192,9 @@ func (u *User) GetRecords(t WorkoutType) (*WorkoutRecord, error) {
 			Where("user_id = ?", u.ID).
 			Where("type = ?", t).
 			Select("workouts.id as id", v+" as value", "workouts.date as date").
+			Order(v + " DESC").
```

Comment:
This should not be necessary; can you explain what you see in your data to add this to the query? (Including the Group and Limit)

Reply:
Eg. the query for `total_distance` for `type = 'running'` is:

```sql
SELECT workouts.id as id,max(total_distance) as value,workouts.date as date
FROM `workouts` join map_data on workouts.id = map_data.workout_id
WHERE user_id = 2 AND type = 'running';
```

We could improve the whole thing to fetch records for all types at once, with this query:

```sql
SELECT workouts.id as id,type, max(max_speed) as value,workouts.date as date
FROM `workouts` join map_data on workouts.id = map_data.workout_id
WHERE user_id = 2
GROUP BY type
```

This would return the best top speed for every workout type (eg. running, walking, cycling), with the workout ID and date.

Reply:
Without the order it just picks the first uploaded workout instead of the workout with the highest "value":

Without order:

![image](https://github.com/jovandeginste/workout-tracker/assets/1283854/096a1409-9ad7-4813-983f-67846651b828)

With order:

![image](https://github.com/jovandeginste/workout-tracker/assets/1283854/c0a729fb-9847-4d5b-a72b-8df089fd0f27)

The `group` is needed to satisfy postgresql "everything needs to be aggregate or grouped" requirement because workout <-> map_data isn't strictly 1-1.

The limit might be unnecessary, I'm not sure how gorm's `Scan` works exactly.

Reply:
Is this specifically when using postgres? Because that's not what I see when using SQLite...

Reply:
Does the logic change when the relation is strictly 1-on-1? Does it work as you would expect when you rebase on the current master branch?

I also notice that I made an error in the queries for average: the records should show `max(average...)` instead of `avg(average...)`; this I will change later.

Reply:
I'm seeing the same behavior with a fresh sqlite instance.

Uploaded a short ride first, then a longer ride. The "max distance" ride shown is from the short ride that was uploaded first

Reply:
That's weird, I see correct data:

![image](https://github.com/jovandeginste/workout-tracker/assets/3170771/48ae008a-4b4a-46bd-9956-cacb8718cfa6)